### PR TITLE
Avoid ConnectingFullmesh trap on advertise_uri change

### PR DESCRIPTION
**Pull request by rosik**
On restart cartrige composes replication config based on clusterwide
configuration. Restarting an instance with a different `advertise_uri`
results in `box.cfg({replication = old_uri, quorum = 1})`.
And since the `old_uri` is unavailable, it gets stuck.

This patch changes the way of replication config generation: present
`advertise_uri` is used for the current instance instead of an old URI.

I didn't forget about

- [x] Tests
- [x] Changelog
- [x] Documentation (unnecessary)

### CODE REVIEW

File: test/integration/reboot_test.lua [hunk 1]
```diff
@@ -198,3 +198,15 @@ function g.test_invalid_config()
...
+function g.test_advertise_change()
+    local master = g.cluster.main_server
+    master:stop()
+
+    master.env['TARANTOOL_ADVERTISE_URI'] = '127.0.0.1:13310'
+    master.net_box_uri = '127.0.0.1:13310'
+    master:start()
+
+    helpers.wish_state(master, 'RolesConfigured')
+
```
Comment (olegrok) [inline test/integration/reboot_test.lua:211]: Redundant line

File: test/integration/reboot_test.lua [hunk 2]
```diff
@@ -198,3 +198,15 @@ function g.test_invalid_config()
...
+function g.test_advertise_change()
+    local master = g.cluster.main_server
+    master:stop()
+
+    master.env['TARANTOOL_ADVERTISE_URI'] = '127.0.0.1:13310'
+    master.net_box_uri = '127.0.0.1:13310'
+    master:start()
+
+    helpers.wish_state(master, 'RolesConfigured')
```
Comment (olegrok) [inline test/integration/reboot_test.lua:210]: It would be great to specify timeout here

File: test/integration/reboot_test.lua [hunk 3]
```diff
@@ -198,3 +198,15 @@ function g.test_invalid_config()
         " not in clusterwide config, no idea what to do now"
     })
 end
+
+function g.test_advertise_change()
+    local master = g.cluster.main_server
```
Comment (olegrok) [inline test/integration/reboot_test.lua:203]: The test is seems to be OK. But I think you can add more complex case when master is stopped, leader is changed and some DML or even DDL actions are made.
After that master is restarted with different advertise_uri.

File: cartridge/confapplier.lua [hunk 4]
```diff
@@ -381,18 +382,11 @@ local function boot_instance(clusterwide_config)
...
-        -- Due to the bug box_opts.replication_connect_quorum was ignored
-        -- and box.cfg used to hang
-        table.remove(
-            box_opts.replication,
-            utils.table_find(
-                box_opts.replication,
-                pool.format_uri(vars.advertise_uri)
-            )
+            topology_cfg, replicaset_uuid,
+            -- Workaround for https://github.com/tarantool/tarantool/issues/3760
```
Comment (olegrok) [inline cartridge/confapplier.lua:386]: Seems patch has been already cherry-picked https://github.com/tarantool/tarantool/issues/3760

Not sure we should support some workarounds.

Comment (rosik) [reply cartridge/confapplier.lua:386]: I've removed stale comment, but I'd like to preserve the workaround - it doesn't affect anything.

File: test/integration/reboot_test.lua (hunk 1 above)
Comment (rosik) [reply test/integration/reboot_test.lua]: Fixed

File: test/integration/reboot_test.lua (hunk 2 above)
Comment (rosik) [reply test/integration/reboot_test.lua]: Done

File: test/integration/reboot_test.lua (hunk 3 above)
Comment (rosik) [reply test/integration/reboot_test.lua:203]: I've added a ddl operation. There is no other instances  to change leader. Rebootstrap is already tested in `integration.rejoin.test_rejoin_*`

Comment (olegrok) [reply test/integration/reboot_test.lua:203]: What will happen if several instances will change its advertise-uri?

Comment (rosik) [reply test/integration/reboot_test.lua:203]: Unless you've specified `replication_connect_quorum = 0` it'll stay in ConnectingFullmesh.

Comment (rosik) [reply test/integration/reboot_test.lua:203]: (It's intentional)

Comment (olegrok) [reply test/integration/reboot_test.lua:203]: Well, but seems it should be documented behaviour

**Review by olegrok — APPROVED** [review]

**Pull request merged**